### PR TITLE
feat(sql, rust, python): SQL CTE's

### PR DESCRIPTION
**Pull request by universalmind303**

### CODE REVIEW

File: polars/polars-sql/src/context.rs [hunk 1]
```diff
@@ -259,7 +287,7 @@ impl SQLContext {
         args: &[FunctionArg],
     ) -> PolarsResult<(String, LazyFrame)> {
         let tbl_fn = name.0.get(0).unwrap().value.as_str();
-
+        println!("executing tbl function");
```
Comment (ritchie46) [inline polars/polars-sql/src/context.rs:290]: I think this a leftover?

File: polars/Makefile [hunk 2]
```diff
@@ -16,7 +16,8 @@ check:
 	    -p polars-arrow \
 	    -p polars-time \
 	    -p polars-error \
-	    -p polars-ops
+	    -p polars-ops \
+			-p polars-sql
```
Comment (ritchie46) [inline polars/Makefile:20]: Could you dedent this?

File: polars/Makefile [hunk 3]
```diff
@@ -43,6 +45,7 @@ test:  ## Run tests
 	    -p polars-time \
 	    -p polars-utils \
 	    -p polars-row \
+			-p polars-sql \
```
Comment (ritchie46) [inline polars/Makefile:48]: Could you dedent this?

**Review by ritchie46 — COMMENTED** [review]
This is great! I think we are now SQL complete enough to at least add this more prevalent on our README. Awesome effort @universalmind303.

I have left a few minor comments.

**Comment by universalmind303** [comment]
@ritchie46 looks like that same unrelated python test is failing again.

**Pull request merged**